index.d.ts

```diff
@@ -0,0 +1,486 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+// TypeScript Version: 2.3.2
+// With great contributions to @akim95 on github
+
+declare namespace Realm {
+    /**
+     * PropertyType
+     * @see { @link https://realm.io/docs/javascript/latest/api/Realm.html#~PropertyType }
+     */
+    type PropertyType = string | 'bool' | 'int' | 'float' | 'double' | 'string' | 'data' | 'date' | 'list';
+
+    /**
+     * ObjectSchemaProperty
+     * @see { @link https://realm.io/docs/javascript/latest/api/Realm.html#~ObjectSchemaProperty }
+     */
+    interface ObjectSchemaProperty {
+        type: PropertyType;
+        objectType?: string;
+        default?: any;
+        optional?: boolean;
+        indexed?: boolean;
+    }
+
+    // properties types
+    interface PropertiesTypes {
+        [keys: string]: PropertyType | ObjectSchemaProperty;
+    }
+
+    /**
+     * ObjectSchema
+     * @see { @link https://realm.io/docs/javascript/latest/api/Realm.html#~ObjectSchema }
+     */
+    interface ObjectSchema {
+        name: string;
+        primaryKey?: string;
+        properties: PropertiesTypes;
+    }
+
+    /**
+     * ObjectClass
+     * @see { @link https://realm.io/docs/javascript/latest/api/Realm.html#~ObjectClass }
+     */
+    interface ObjectClass {
+        schema: ObjectSchema;
+    }
+
+    /**
+     * ObjectType
+     * @see { @link https://realm.io/docs/javascript/latest/api/Realm.html#~ObjectType }
+     */
+    interface ObjectType {
+        type: ObjectClass;
+    }
+
+    /**
+     * realm configuration
+     * @see { @link https://realm.io/docs/javascript/latest/api/Realm.html#~Configuration }
+     */
+    interface Configuration {
+        encryptionKey?: any;
```

Comment:
`encryptionKey?` should be `ArrayBuffer | ArrayBufferView` or better yet just `Int8Array`.